src/components/donation/Steps/CurrentStep.tsx

```diff
@@ -19,21 +19,27 @@ export default function CurrentStep({ config }: Props) {
   }
 
   if (state.step === "splits") {
-    return <Splits {...state} />;
+    return (
+      <Splits
+        {...state}
+        liquidSplitPct={state.liquidSplitPct ?? config?.liquidSplitPct}
```

Comment:
this leads to another problem, now donation state has priority over config state

bug repro:
1. go to widget configurer
2. (akin to user browsing the preview) , try completing donation form and split (split now saved in state)
3. go back to split step
4. user tries to modify the widget split
5. update snippet

expected: wigdet split should reflect but no

Reply:
Dear lord.. will fix this and thoroughly test this time.

Reply:
So we need to make it so that `state.liquidSplitPct` always has precedence except when a new `config.liquidSplitPct` is introduced.
One simple way to this is to always use `state.liquidSplitPct` in `Splits` component and update this state value when `config.liquidSplitPct` changes. But this would require calling `dispatch(setSplits(config.liquidSplitPct))` when initializing the donate widget form, which would be a problem with the current setup as all `slices/donation` actions immediately change the step type when called. The solution to this problem would be to make actions "dumber", in the sense that they just perform a state update, but leave changing the step type to some other action - this ties into our recent discussion about creating `goToNextStep`, but if you still think this is not desirable, we can use the existing `setStep`.
This would mean rewriting:
```ts
dispatch(setSplits(value));
```
to:
```ts
dispatch(setSplits(value));
dispatch(setStep("step-value");
```

What do you think?

Reply:
@SovereignAndrey @ap-justin I need your opinion on the proposal from my previous comment, as the issue is a blocker to finishing the ticket.

Reply:
@misicnenad,
 kindly look at this sample solution https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2817/files

let me know if there's still bug

Reply:
@ap-justin I checked out the draft solution you provided, it works as described in the PR descr:
> dominant donation split over widget split (on donation), but widget split is dominant on preview

In my draft PR for the same problem (PR https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2819), I tried to make the widget split use whichever value was the most recently changed (config VS actual split slider).
Seeing your assumptions when creating your PR, I realized we both understood the problem (and the solution) _differently_.

I think we're missing a piece of the puzzle though - _we don't actually know what is the expected behavior here._

Gonna call in the cavalry here @SovereignAndrey - need your input here ser - what is the expected behavior when widget config's liquid % value conflicts with the user-selected slider value in the following cases (gonna list my assumptions and questions for you together):
1. Admin Donation Form Builder - user sets default (config) split value - widget preview updates to the new value - possible cases are:
   a. The user goes to the next step and then returns, the split value should still be the default split value.
   b. The user changes the split value manually, then they go to the next step and go back, **which value should be
        displayed/selected in the Splits screen? The default config value OR the manually set value?**
   c. Related to the previous, if the user changes the split manually and goes to the next step, **which split value should be used for calculations, the default config value OR the new manually set value?**
2. Donation Widget - endowment set default (config) split value and enabled changing it for donators - possible cases are: 
   a. The user goes to the next step and then returns, the split value should still be the default split value.
   b. The user changes the split value manually, then they go to the next step and go back, **which value should be
        displayed/selected in the Splits screen? The default config value OR the manually set (user's) value?** It makes sense to use the user's value here.
   c. Related to the previous, if the user changes the split manually and goes to the next step, **we should continue using the new manually set value?**
3. Regular Donation Form - there's no option for an endowment to set the default split value, so no problems here.

Reply:
@misicnenad, #2817 is based on how I, as a user, expects the UI to function

> 1. Admin Donation Form Builder 

 The configurer should always dictate what the preview would look like. I could play around the widget preview (change it's state different from the config), but ultimately when I click `[update snippet]`, preview should reflect the configuration. Preview state is irrelevant, as it's just a preview and won't push through to an actual donation. 
 
 > 2. Donation Widget 
 
 Donation widget should also reflect its configuration (e.g. split:`99`), and if configuration allows for me to choose differently (e.g. split is not disabled) ,I would want my intent to be reflected (persisted). Most of the time, I understand that If I refresh a page/form/flow - it would reset to its initial state (initial config of widget in this case)
 
 > 3 Donate page
 
 as usual as no widget config is applied

Reply:
> #2817 is based on how I, as a user, expects the UI to function (might be different than your preferences sers , @misicnenad @SovereignAndrey
> 
> > 1. Admin Donation Form Builder 
> 
>  The configurer should always reflect what I configure. I could play around the widget preview (change it's state different from the config), but ultimately when I click `[update snippet]`, preview should reflect my configuration. Preview state is secondary, as it's just a preview and won't push through to an actual donation. 
>  
>  > 2. Donation Widget 
>  
>  Donation widget should also reflect its configuration (e.g. split:`99`), and if configuration allows for me to choose differently (e.g. split is not disabled) ,I would want my intent to be reflected (persisted). Most of the time, I understand that If I refresh a page/form/flow - it would reset to its initial state (initial config of widget in this case)
>  
>  > 3 Donate page
>  
>  as usual as no widget config is applied
>  
>  
> 

@misicnenad this reflects how I would expect the widget to function as well with regards to the config and changes made by a user.

Reply:
Merged Justin's PR, see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2802/commits/1c26624aa084b4a7ba5991d33a2427ab25576b41